tests/boot/test_mcuboot/testcase.yaml

```diff
@@ -0,0 +1,17 @@
+common:
+    sysbuild: True
+    harness: console
+    harness_config:
+      type: one_line
+      regex:
+        - "I: Starting bootloader"
+        - "Launching primary slot application on (.*)"
+        - "Secondary application ready for swap, rebooting"
+        - "I: Starting swap using (.*)"
+        - "Swapped application booted on (.*)"
+tests:
+  boot.mcuboot:
+    tags: mcuboot
+    platform_allow: frdm_k64f mimxrt1060_evk
```

Comment:
Is there any reason for this list being only 2 boards? There are plenty of other boards that support MCUboot.

Reply:
Exactly, `platform_allow` should only been used when there is no way a test can run on anything outside from this list. It mustn't be used to indicate "I tested it on these 2 platforms and it work for me, don't know about the rest".  I expanded it with nrf52840dk_nrf52840 and the test passed. Keeping platform_allow here will prevent others from being able to run this test. Please, propose some other way of filtering if required.

Reply:
I agree that `platform_allow` is too restrictive here- however, twister is unable to filter the mcuboot test case in it's current state (since mcuboot requires a boot partition, and devices without that partition will fail to complete CMake configuration)

@tejlmand made me aware of #41302, which would allow twister to filter the MCUBoot sample correctly, by only running some configuration steps. However, as far as I can see twister would need an update to support filtering with only specific cmake modules. I didn't want to over complicate the change set in this PR, so I have not added support for filtering with modules here. My plan was to follow up this PR with one enabling filtering using cmake modules, which will allow the mcuboot test to run on a variety of platforms.

Reply:
I believe you're more then welcome to add support of #41302 into twister. I was also aware of it but didn't have time yet to add support for it. Since you already dived into twister code and build stages I believe you are well equipped for this task ;) 
IMO (and I know @tejlmand also shares this opinion) it would be super useful in twister: allowing us to save time building test which will end skipped in the end and making this filtration available without full cmake (which can fail).

Reply:
I'm not convinced that some more sophisticated mechanism will always works. MCUboot basically requires proper flash partitioning, flash driver which supports write-block-size <= 32 B, supported by it architecture (for instance CortexM3...). Even that platform fulfil all these requirement MCUboot might be not the case for the board (for instance some flash memories are really weird - therefore incompatible, different bootloader is expected etc.)